api/src/test/java/org/openmrs/BaseFormRecordableOpenmrsDataTest.java

```diff
@@ -0,0 +1,158 @@
+		impl.setFormField(ns, path);
+		assertEquals(ns, impl.getFormFieldNamespace());
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#getFormFieldNamespace()
+	 */
+	@Test
+	public void getFormFieldNamespace_shouldReturnTheNamespaceForAFormFieldThatHasNoPath() throws Exception {
+		final String ns = "my ns";
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		impl.setFormField(ns, null);
+		assertEquals(ns, impl.getFormFieldNamespace());
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#getFormFieldPath()
+	 */
+	@Test
+	public void getFormFieldPath_shouldReturnNullIfThePathIsNotSpecified() throws Exception {
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		impl.setFormField("my ns", "");
+		assertNull(impl.getFormFieldPath());
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#getFormFieldPath()
+	 */
+	@Test
+	public void getFormFieldPath_shouldReturnTheCorrectPathForAFormFieldWithANamespace() throws Exception {
+		final String ns = "my ns";
+		final String path = "my path";
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		impl.setFormField(ns, path);
+		assertEquals(path, impl.getFormFieldPath());
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#getFormFieldPath()
+	 */
+	@Test
+	public void getFormFieldPath_shouldReturnThePathForAFormFieldThatHasNoNamespace() throws Exception {
+		final String path = "my path";
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		impl.setFormField("", path);
+		assertEquals(path, impl.getFormFieldPath());
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#setFormField(String,String)
+	 */
+	@Test
+	public void setFormField_shouldRejectANamepaceAndPathCombinationLongerThanTheMaxLength() throws Exception {
+
+		final String ns = StringUtils.repeat("x", 255);
+		final String path = "";
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		assertThrows(APIException.class, () -> impl.setFormField(ns, path));
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#setFormField(String,String)
+	 */
+	@Test
+	public void setFormField_shouldRejectANamepaceContainingTheSeparator() throws Exception {
+		final String ns = "my ns" + FORM_NAMESPACE_PATH_SEPARATOR;
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		assertThrows(APIException.class, () -> impl.setFormField(ns, ""));
+	}
+
+	/**
+	 * @see BaseFormRecordableOpenmrsDataImpl#setFormField(String,String)
+	 */
+	@Test
+	public void setFormField_shouldRejectAPathContainingTheSeparator() throws Exception {
+		final String path = FORM_NAMESPACE_PATH_SEPARATOR + "my path";
+		BaseFormRecordableOpenmrsDataImpl impl = new BaseFormRecordableOpenmrsDataImpl();
+		assertThrows(APIException.class, () -> impl.setFormField("", path));
+	}
+
+	public class BaseFormRecordableOpenmrsDataImpl extends BaseFormRecordableOpenmrsData {
```

Comment:
Why not just use an `Obs`? That'd be the more backward compatible approach.

Reply:
I prefer create a test class for this  (since is not to complicated) in order to the test be more independent.

Reply:
It feels safer doing tests with real existing classes. If independency is needed, then i would do such tests in addition to those that deal with the real existing classes, not minus. 😊

Reply:
FWIW, i have ever had a case where a dummy test class masked a bug in a real class. 😊

Reply:
@icrc-loliveira... you'll have to use `Obs` ;-)

See [here](https://github.com/openmrs/openmrs-core/pull/3357#issuecomment-660128032):
>@mks-d i still have a problem with replacing obs with a dummy BaseFormRecordableOpenmrsDataImpl class in the tests.